src/assets/main.js

```diff
@@ -91,8 +91,27 @@ const routes = [
     },
 ];
 
+const vuetify = new Vuetify({
+    // Initialize with pre-calculated theme to prevent flash of wrong theme
+    theme: {
+        dark: window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches,
+    },
```

Comment:
Maybe we can reuse some of the code.

```suggestion
    theme: {
        dark: util.theme._resolve() == 'dark',
    },
```